platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/appcomm/LatestCompleteStateNotifier.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.components.appcomm;
+
+import com.swirlds.common.notification.NotificationEngine;
+import com.swirlds.platform.consensus.ConsensusConstants;
+import com.swirlds.platform.listeners.StateWriteToDiskCompleteListener;
+import com.swirlds.platform.listeners.StateWriteToDiskCompleteNotification;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import com.swirlds.platform.state.signed.StateSavingResult;
+import com.swirlds.platform.system.state.notifications.NewSignedStateListener;
+import com.swirlds.platform.system.state.notifications.NewSignedStateNotification;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * This component responsible for notifying the application of latest complete state
+ */
+public class LatestCompleteStateNotifier {
+    private final NotificationEngine notificationEngine;
+
+    /** The round of the latest state provided to the application */
+    private long latestStateProvidedRound = ConsensusConstants.ROUND_UNDEFINED;
+
+    /**
+     * Create a new instance
+     *
+     * @param notificationEngine the notification engine
+     */
+    public LatestCompleteStateNotifier(@NonNull final NotificationEngine notificationEngine) {
+        this.notificationEngine = Objects.requireNonNull(notificationEngine);
+    }
+
+    /**
+     * Notify the application that a state has been saved to disk successfully
+     *
+     * @param stateSavingResult the result of the state saving operation
+     */
+    public void stateSavedToDisk(@NonNull final StateSavingResult stateSavingResult) {
```

Comment:
this should move, it is unrelated to the latest complete state

Reply:
done. Moved to platform.